### PR TITLE
Added SE 2013 TAS support and basic missinginfo support

### PR DESCRIPTION
**Pull request by TheProJamer**
All singleplayer games supported except Half-Life Source, which seemed to have rather extreme auto-strafe problems. I'm leaving that until I work out the issue, or someone who can help me does help me.
Added TAS support to:
- Half-Life 2 (build 2257546)
- Half-Life 2 Episode One and Two (build 2257546)
- Half-Life 2 Lost Coast (build 2257546)
- Portal (build 1910503)
Added basic SPT support to:
- Missing Information 1.6 (build 4104)
- Missing Information 1.4.7 (build 4044)

### CODE REVIEW

File: spt/OrangeBox/patterns.hpp [hunk 1]
```diff
@@ -355,22 +363,6 @@ namespace Patterns
 			"xxxxxxxxxxxx????xxxxxxxxxxxxxxxxxx"
 		},
 
-		{
```
Comment (YaLTeR) [inline spt/OrangeBox/patterns.hpp]: Why was this removed?

Comment (TheProJamer) [reply spt/OrangeBox/patterns.hpp]: I have no idea.

Comment (YaLTeR) [reply spt/OrangeBox/patterns.hpp]: Add it back?

Comment (TheProJamer) [reply spt/OrangeBox/patterns.hpp]: After further examination, I've come to the conclusion that I removed it because the same two patterns are directly above it.

Comment (YaLTeR) [reply spt/OrangeBox/patterns.hpp]: Fair enough.
